sqs_s3_logger/lambda_function.py

```diff
@@ -26,23 +26,41 @@ def handler(event, context):
     s3.meta.client.head_bucket(Bucket=b_name)  # Check that the bucket exists.
     b = s3.Bucket(b_name)
     msgs = read_queue(q)
-    dump_messages_to_file(msgs, temp_filepath)
+    written = dump_messages_to_file(msgs, temp_filepath)
     b.upload_file(temp_filepath, '{}.txt'.format(datetime.datetime.now()))
 
+    failed_deletions = batch_delete_messages(q, written)
+    if failed_deletions:
+        LOGGER.info('Deletion of {} messages failed'.format(len(failed_deletions)))
+
+
+def batch_delete_messages(queue, messages):
+    deletion_results = []
+    for i in range(0, len(messages), 10):
```

Comment:
I could not find any documentation on the upper limit of batch sizes anywhere, so I assumed it is the same as when reading (it also works for me that way).